src/cli/utils/utils.go

```diff
@@ -24,34 +24,53 @@ const (
 	Wildcard = "*"
 )
 
-func GetAccountAndConnect(ctx context.Context) (repository.Repository, *account.Account, error) {
+func GetAccountAndConnect(ctx context.Context) (repository.Repository, *storage.Storage, *account.Account, error) {
 	cfg, err := config.GetConfigRepoDetails(ctx, true, nil)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	repoID := cfg.RepoID
 	if len(repoID) == 0 {
-		repoID = events.RepoID
+		repoID = events.RepoIDNotFound
 	}
 
 	r, err := repository.Connect(ctx, cfg.Account, cfg.Storage, repoID, options.Control())
 	if err != nil {
-		return nil, nil, clues.Wrap(err, "Failed to connect to the "+cfg.Storage.Provider.String()+" repository")
+		return nil, nil, nil, clues.Wrap(err, "Failed to connect to the "+cfg.Storage.Provider.String()+" repository")
 	}
 
-	s3Config, errS3Config := cfg.Storage.S3Config()
-	m365Config, errm365Config := cfg.Account.M365Config()
+	return r, &cfg.Storage, &cfg.Account, nil
+}
+
+func AccountConnectAndWriteRepoConfig(ctx context.Context) (repository.Repository, *account.Account, error) {
+	r, stg, acc, err := GetAccountAndConnect(ctx)
+	if err != nil {
+		logger.CtxErr(ctx, err).Info("Failed to  get and connect account")
```

Comment:
Per [logging standards](https://github.com/alcionai/corso/blob/main/src/pkg/logger/example_logger_test.go), avoid prefixes like "failed to" in log messages.
```suggestion
		logger.CtxErr(ctx, err).Info("connecting to repo")
```